src/Microsoft.AspNetCore.TestHost/WebSocketClient.cs

```diff
@@ -87,6 +87,7 @@ private class RequestState : IDisposable, IHttpWebSocketFeature
         {
             private readonly IHttpApplication<Context> _application;
             private TaskCompletionSource<WebSocket> _clientWebSocketTcs;
+            private CancellationTokenRegistration _cancellationTokenRegistration;
```

Comment:
It isn't used but it is referenced preventing GC. I wasn't sure whether I would call Dispose on it, but that would possibly happen before all references to `_clientWebSocketTcs.Task` and the `CancellationTokenSource` the client might have are gone and hence break this fix. Notice `_clientWebSocketTcs` isn't being disposed neither.